BackUp: Previous Development

ThomasNaderBMW opened the pull request:
Bitte einfach kommentieren :)

ThomasNaderBMW committed:
Update osi_vehicle.proto

SergejHolzmannBMW commented:
Aus meiner Sicht fehlen noch folgende Informationen:

VehicleBasics:
- color
- number plate

DriverActions:
- Enum mit möglichen Tätigkeiten (fahren, Zeitung lesen, schlafen, telefonieren, trinken, schimpfen,...)

ThomasNaderBMW committed:
Interface to communicate the status of simulation-participants

ThomasNaderBMW committed:
Update osi_control

ThomasNaderBMW committed:
Update osi_control

ThomasNaderBMW committed:
Osi_graphics: First Draft

Feel free to comment!

TKUsaneers commented:
Für unser AR-HUD fehlen noch:
1. Kopfposition bzw. Augpunkt des Fahrers
2. Navigationsinformation
2.1 Geplanter Streckenverlauf
2.2 Anstehende Maneuver mit Position, Maneuvertyp (z.B. Spurwechsel, Abbiegen), und eventueller Zusatzinformation (z.B. Straßennamen, Zielspur des Spurwechsels)
3. FAS-Funktionen
3.1 Regelobjekt des ACC
3.2 Erkannte Engstellen
3.3 Aktivierungssignal inklusive Level
3.4 Deaktivierungssignal inklusive Level und Counter/Timer
3.5 Spurverlassenswarnung inklusive betroffene Seite
3.6 WBC: Gefährdendes Objekt, sowie Ausweich-Trajektorie (falls Level 3/4)
3.7 HOR inklusive Level 
3.8 TOR inklusive Level 
3.9 Nothalt inlusive Distanz bis zum Stillstand und Trajektorie
3.10 Abstandswarnung inklusive erkanntem Objekt/Hinderniss
4. Ist es möglich die Trajektorie nicht nur als Ankerpunkt zu bekommen, sondern, wie die normalen Spuren, als Liste von Koordinaten?

ManuelKugelmann commented:
osi_vehicle.proto is missing includes:

import "osi_version.proto";

(and maybe import "osi_object.proto"; for LightState, although it is wrapped in MovingObject_VehicleClassification_LightState)

ManuelKugelmann commented:
osi_vehicle.proto Vehicle has duplicate field ids:

    optional InterfaceVersion version = 1;
    optional VehicleBasics vehicle_basics = 1;

    optional LightState vehicle_light_state = 7;
    repeated VehicleAutomatedDrivingFunction vehicle_automated_driving_function = 7;

ManuelKugelmann commented:
Vehicle LightState is not properly defined - same as MovingObject_VehicleClassification_LightState ?

ThomasNaderBMW committed:
Corrected the mistakes mentioned by @TheDraguun

Thx for the input!

ThomasNaderBMW commented:
> Vehicle LightState is not properly defined - same as MovingObject_VehicleClassification_LightState ?

@TheDraguun I corrected the mistakes you found. Light_state is wrapped in osi_object, so should be included now. Thx for your comments!
So far it is the same like MovingObject_VehicleClassification_LightState, if this is not enough there are two options we can consider:
- Extension of osi_object (LightState) by additional signals we think are missing
- Or define an own Light-Message

ManuelKugelmann commented:
Question: where will the Vehicle message be placed? 
As member of SensorView like GroundTruth?
Or into a new root message type?



> @TheDraguun I corrected the mistakes you found. Light_state is wrapped in osi_object, so should be included now. Thx for your comments!
 
The LightState message type in Vehicle and Intentions still needs to be fully named:
 optional LightState vehicle_light_state = 8; 
should read
 optional MovingObject.VehicleClassification.LightState vehicle_light_state = 8;

ThomasNaderBMW committed:
Draft. Under construction..

ThomasNaderBMW committed:
Corrected LightState to MovingObject.VehicleClassification.LightState

ThomasNaderBMW commented:
> Question: where will the Vehicle message be placed?
> As member of SensorView like GroundTruth?
> Or into a new root message type?
> 
> > @TheDraguun I corrected the mistakes you found. Light_state is wrapped in osi_object, so should be included now. Thx for your comments!
> 
> The LightState message type in Vehicle and Intentions still needs to be fully named:
> optional LightState vehicle_light_state = 8;
> should read
> optional MovingObject.VehicleClassification.LightState vehicle_light_state = 8;

You are right, I corrected it.

ThomasNaderBMW commented:
osi_vehicle.proto should be syntactical correct. I compiled it by using the protoc-compiler. For interested:
https://developers.google.com/protocol-buffers/docs/cpptutorial#why-use-protocol-buffers

ThomasNaderBMW committed:
Update osi_control

ThomasNaderBMW committed:
Bigger change to osi_control

Now it is also possible to send commands regarding the target state to one ore more external application(s).

ThomasNaderBMW committed:
Rename osi_control to osi_control.proto

ThomasNaderBMW committed:
Rename osi_graphics to osi_graphics.proto

ThomasNaderBMW committed:
Update osi_graphics.proto

SergejHolzmannBMW commented on osi_graphics.proto:
```diff
@@ -0,0 +1,167 @@
+    //
+    // Here is the main information about how the graphic-engine is configured.
+    // E.g.: Place of the camera, screen-configuration, avatar information, ...
+    //
```
ADD: Visualization mode (VR, Single screen, Multi screen)

SergejHolzmannBMW commented on osi_graphics.proto:
```diff
@@ -0,0 +1,167 @@
+//
+message ConfigurationScreen
+{
+    // Some open points:
```
ADD: id
Please consider the nDisplay configuration: 
https://docs.unrealengine.com/en-US/Engine/Rendering/nDisplay/Configuration/index.html

ThomasNaderBMW replied:
Komm da bitte nochmal auf mich zu.

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
Update osi_vehicle.proto

ThomasNaderBMW committed:
Update osi_vehicle.proto

ThomasNaderBMW committed:
Update osi_control.proto

ManuelKugelmann commented:
Question: where will the Vehicle, Control and Graphics messages be placed?
As member of SensorView like GroundTruth?
Or is it a new root message type?

ManuelKugelmann commented:
@ThomasNaderBMW Any news on how the new message types will be published?

ThomasNaderBMW committed:
Update osi_control.proto

ThomasNaderBMW committed:
Update osi_control.proto

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
Update osi_control.proto

ThomasNaderBMW committed:
Update osi_control.proto

ThomasNaderBMW committed:
Updated Code-Styling

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
First draft that could work.

ThomasNaderBMW committed:
Update osi_vehicle.proto

ThomasNaderBMW committed:
under construction

ThomasNaderBMW committed:
Ready to test.

ThomasNaderBMW committed:
Update osi_control.proto

ThomasNaderBMW committed:
Ready to test.

ManuelKugelmann commented:
osi_vehicle.proto just contains a link to the real code:
https://raw.githubusercontent.com/OpenSimulationInterface/open-simulation-interface/7a26b8b87e43d0ec854246d1a6587a856111a4f3/osi_vehicle.proto
Wouldn't it be nicer if this branch were a complete OSI3 definition without external links?

ThomasNaderBMW commented:
> osi_vehicle.proto just contains a link to the real code:
> https://raw.githubusercontent.com/OpenSimulationInterface/open-simulation-interface/7a26b8b87e43d0ec854246d1a6587a856111a4f3/osi_vehicle.proto
> Wouldn't it be nicer if this branch were a complete OSI3 definition without external links?

What do you mean by external links?
The include of osi_occupant?

ManuelKugelmann commented:
> > osi_vehicle.proto just contains a link to the real code:
> > https://raw.githubusercontent.com/OpenSimulationInterface/open-simulation-interface/7a26b8b87e43d0ec854246d1a6587a856111a4f3/osi_vehicle.proto
> > Wouldn't it be nicer if this branch were a complete OSI3 definition without external links?
> 
> What do you mean by external links?
> The include of osi_occupant?

The link in osi_vehicle.proto that replaces its content. I took this branch and ran the proto compiler on it - it spit an error for osi-vehicle.proto, I had to follow the link, click a few times to get the code  and paste it into the file. All this  (imho unneccessary) indirection could be skipped, if the file just contained the code. It also would prevent a (false) merge conflict showing for osi_vehicle.proto.

ThomasNaderBMW commented:
> > > osi_vehicle.proto just contains a link to the real code:
> > > https://raw.githubusercontent.com/OpenSimulationInterface/open-simulation-interface/7a26b8b87e43d0ec854246d1a6587a856111a4f3/osi_vehicle.proto
> > > Wouldn't it be nicer if this branch were a complete OSI3 definition without external links?
> > 
> > 
> > What do you mean by external links?
> > The include of osi_occupant?
> 
> The link in osi_vehicle.proto that replaces its content. I took this branch and ran the proto compiler on it - it spit an error for osi-vehicle.proto, I had to follow the link, click a few times to get the code and paste it into the file. All this (imho unneccessary) indirection could be skipped, if the file just contained the code. It also would prevent a (false) merge conflict showing for osi_vehicle.proto.

Maybe it makes sense to talk directly. Please send me an E-Mail ;)

ManuelKugelmann commented:
> > > > osi_vehicle.proto just contains a link to the real code:
> > > > https://raw.githubusercontent.com/OpenSimulationInterface/open-simulation-interface/7a26b8b87e43d0ec854246d1a6587a856111a4f3/osi_vehicle.proto
> > > > Wouldn't it be nicer if this branch were a complete OSI3 definition without external links?
> > > 
> > > 
> > > What do you mean by external links?
> > > The include of osi_occupant?
> > 
> > 
> > The link in osi_vehicle.proto that replaces its content. I took this branch and ran the proto compiler on it - it spit an error for osi-vehicle.proto, I had to follow the link, click a few times to get the code and paste it into the file. All this (imho unneccessary) indirection could be skipped, if the file just contained the code. It also would prevent a (false) merge conflict showing for osi_vehicle.proto.
> 
> Maybe it makes sense to talk directly. Please send me an E-Mail ;)

Nevermind - the problem is only subjective and concerns just the meta level, thus it is not worth the time for extended discussion. Resolving the missing code is straight forward and your approach might have benefits to other collaborators that I'm missing.

ManuelKugelmann commented:
Found a compiler error when trying to build C++ code:
the `Notification `message has a duplicate definition: It is defined and used in osi_vehicle.proto and in osi_control.proto.

I will do a local quickfix by changing osi_vehicle.proto to `import "osi_control.proto";` and deleting the Notification definition.

ThomasNaderBMW committed:
Update osi_control.proto

ThomasNaderBMW commented:
> Found a compiler error when trying to build C++ code:
> the `Notification `message has a duplicate definition: It is defined and used in osi_vehicle.proto and in osi_control.proto.
> 
> I will do a local quickfix by changing osi_vehicle.proto to `import "osi_control.proto";` and deleting the Notification definition.

Solved the problem by moving all external control messagedefinitions into the control message. Is that working for you?

ManuelKugelmann commented:
> > Found a compiler error when trying to build C++ code:
> > the `Notification `message has a duplicate definition: It is defined and used in osi_vehicle.proto and in osi_control.proto.
> > I will do a local quickfix by changing osi_vehicle.proto to `import "osi_control.proto";` and deleting the Notification definition.
> 
> Solved the problem by moving all external control messagedefinitions into the control message. Is that working for you?

compiles.

Onward to state containers and field setters ...

ThomasNaderBMW committed:
Corrected some bugs

ChristopherKiwus committed:
OSIGraphics: Correcting typing error

Corrects a typing error in a description text.

SergejHolzmannBMW commented on osi_graphics.proto:
```diff
@@ -0,0 +1,375 @@
+        //
+        // Unit: [%]
+        //
+        optional double super_sampling_configuration = 8;
```
SuperSampling ist ein globaler Wert. Umziehen in ScreenConfiguration.

SergejHolzmannBMW commented on osi_graphics.proto:
```diff
@@ -0,0 +1,375 @@
+        //
+        // ToDo: Description.
+        //
+        optional double frustum = 8;
```
Frustum kann kein double sein.

ThomasNaderBMW replied:
Stattdessen int32?

SergejHolzmannBMW commented on osi_graphics.proto:
```diff
@@ -0,0 +1,375 @@
+// 
+// \brief ToDo: Description
+//
+message VirtualRealityConfiguration
```
Flag für "UseHMDCalibration" notwendig.

SergejHolzmannBMW commented on osi_graphics.proto:
```diff
@@ -0,0 +1,375 @@
+    //Kopfposition
+    // OSI-Env (Tageszeitentypen, Cloudspeed, Wettertypen, Cloudshadowintensity, Percipitation, TimeofDay) country_code
+}
+
```
Add: 
- HUD Information
- SPOUT plane

SergejHolzmannBMW reviewed (commented):
Meeting with Christopher Kiwus & Sergej Holzmann, 20.02.2020

SergejHolzmannBMW commented on osi_graphics.proto:
```diff
@@ -0,0 +1,375 @@
+
+import "osi_common.proto";
+import "osi_version.proto";
+
```
Include "osi_environment" in osi_graphics

SergejHolzmannBMW commented on osi_graphics.proto:
```diff
@@ -0,0 +1,375 @@
+
+    //Open:
+    //Puffer?
+    //IsMotionCompensation
```
Short-term: flag in VirtualRealityConfiguration
Long-term: create MotionCompensation message with numeric values from platform module.

ThomasNaderBMW committed:
Delete osi_vehicle.proto

ThomasNaderBMW committed:
Update osi_graphics.proto

ThomasNaderBMW committed:
Update osi_graphics.proto
